mathics/core/expression.py

```diff
@@ -244,11 +244,16 @@ def __hash__(self):
         return hash(("Expression", self._head) + tuple(self._elements))
 
     def __repr__(self) -> str:
+        return "<Expression: %s[%s]>" % (
+            self._head.__repr__(),
+            ", ".join([element.__repr__() for element in self._elements]),
+        )
+
         return "<Expression: %s>" % self
 
     def __str__(self) -> str:
         return "%s[%s]" % (
-            self._head,
+            self._head.__str__(),
```

Comment:
As above, `str(self.head)` is probably more correct. And possibly other places where we have `self._head` we should be using `self.head`.